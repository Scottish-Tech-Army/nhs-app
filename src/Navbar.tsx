import React from "react";
import { NavLink } from "react-router-dom";
import { ReactComponent as HomeIcon } from "./icons/home.svg";
import { ReactComponent as ChecklistIcon } from "./icons/checklist.svg";

function Navbar() {
  return (
    <nav>
      <NavLink
        aria-label="directory"
        to={"/"}
        className={({ isActive }) => (isActive ? "navicon active" : "navicon")}
      >
        <HomeIcon />
        <div>Directory</div>
      </NavLink>
      <NavLink
        aria-label="summary"
        to={"/summary"}
        className={({ isActive }) => (isActive ? "navicon active" : "navicon")}
      >
        <ChecklistIcon />
        <div>Summary</div>
      </NavLink>
    </nav>
  );
}

export default Navbar;
