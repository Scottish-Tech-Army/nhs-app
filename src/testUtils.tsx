import React from "react";
import { render } from "@testing-library/react";
import type { RenderOptions } from "@testing-library/react";
import { configureStore } from "@reduxjs/toolkit";
import type { PreloadedState } from "@reduxjs/toolkit";
import { Provider } from "react-redux";

import type { AppStore, RootState } from "./data/store";
import boxContentsReducer, { initialState } from "./data/BoxContentsSlice";

interface ExtendedRenderOptions extends Omit<RenderOptions, "queries"> {
  preloadedState?: PreloadedState<RootState>;
  store?: AppStore;
}

export function renderWithProvider(
  component: React.ReactElement,
  {
    preloadedState = { boxContents: initialState },
    store = configureStore({
      reducer: { boxContents: boxContentsReducer },
      preloadedState,
    }),
    ...renderOptions
  }: ExtendedRenderOptions = {}
) {

  return {
    store,
    ...render(<Provider store={store}>{component}</Provider>, renderOptions),
  };
}
