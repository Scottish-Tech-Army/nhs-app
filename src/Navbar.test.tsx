import React from "react";

import { screen } from "@testing-library/react";
import { renderWithProvider } from "./testUtils";
import Navbar from "./Navbar";

describe("Navbar", () => {
  it("renders for summary page", () => {
    renderWithProvider(<Navbar />, {
      initialRoutes: ["/summary"],
    });

    expect(screen.getByRole("link", { name: "storage area" })).not.toHaveClass(
      "active"
    );
    expect(screen.getByRole("link", { name: "summary" })).toHaveClass("active");
  });

  it("renders for storage area", () => {
    const { container } = renderWithProvider(<Navbar />, {
      initialRoutes: ["/"],
    });

    expect(screen.getByRole("link", { name: "storage area" })).toHaveClass(
      "active"
    );
    expect(screen.getByRole("link", { name: "summary" })).not.toHaveClass(
      "active"
    );

    expect(container).toMatchSnapshot();
  });

  it("navigates from summary page", async () => {
    const { user, history } = renderWithProvider(<Navbar />, {
      initialRoutes: ["/summary"],
    });

    await user.click(screen.getByRole("link", { name: "summary" }));

    expect(history.location.pathname).toEqual("/summary");

    await user.click(screen.getByRole("link", { name: "storage area" }));

    expect(history.location.pathname).toEqual("/");
  });

  it("navigates from storage area page", async () => {
    const { user, history } = renderWithProvider(<Navbar />, {
      initialRoutes: ["/"],
    });

    await user.click(screen.getByRole("link", { name: "storage area" }));

    expect(history.location.pathname).toEqual("/");

    await user.click(screen.getByRole("link", { name: "summary" }));

    expect(history.location.pathname).toEqual("/summary");
  });
});
