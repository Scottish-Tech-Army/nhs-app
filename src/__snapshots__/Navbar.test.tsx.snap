// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders for directory 1`] = `
<div>
  <nav>
    <a
      aria-current="page"
      aria-label="directory"
      class="navicon active"
      href="/"
    >
      <svg>
        home.svg
      </svg>
      <div>
        Directory
      </div>
    </a>
    <a
      aria-label="summary"
      class="navicon"
      href="/summary"
    >
      <svg>
        checklist.svg
      </svg>
      <div>
        Summary
      </div>
    </a>
    <div
      aria-label="log out"
      class="navicon"
      id="auth-signout"
    >
      <svg>
        logout.svg
      </svg>
      <div>
        Log Out
      </div>
    </div>
  </nav>
</div>
`;
